packages/schema/src/config/build.ts

```diff
@@ -108,5 +108,37 @@ export default defineUntypedSchema({
         }
       }
     },
+  },
+
+  /**
+   * Build time optimization configuration.
+   */
+  optimization: {
+    /**
+     * Tree shake code from specific builds.
+     */
+    treeShake: {
+      /**
+       * Tree shake composables from the server or client builds.
+       *
+       * @example
+       * ```js
+       * treeShake: { server: ['useClientOnlyComposable'] }
+       * ```
+       */
+      composables: {
+        server: {
+          $resolve: async (val, get) => (await get('dev') ? [] : [
+            'onBeforeMount', 'onMounted', 'onBeforeUpdate', 'onRenderTracked', 'onRenderTriggered', 'onActivated', 'onDeactivated', 'onBeforeUnmount'
+          ]).concat(val).filter(Boolean)
+        },
+        client: {
+          $resolve: async (val, get) => (await get('dev') ? [] : [
+            'onServerPrefetch', 'onRenderTracked', 'onRenderTriggered'
+          ]).concat(val).filter(Boolean)
+        }
```

Comment:
while we're defining the API, would it make sense to have an array of objects with name + exporter (e.g. these would all be from 'vue')? That way if we later improve the tree shaking algorithm to remove only ones from a certain import-path (or even integrate with unimport), then we can use a finer-grained hammer.

Reply:
Another great idea! Have implemented, let me know if it needs any further changes